src/lib/eina/eina_sched.c

```diff
@@ -82,3 +83,48 @@ eina_sched_prio_drop(void)
      }
 # endif
 }
+#else
+EAPI void
+eina_sched_prio_drop(void)
+{
+   Eina_Thread pthread_id;
+   int sched_priority;
+
+   pthread_id = eina_thread_self(); //pthread_id = pthread_self();
+
+   sched_priority = GetThreadPriority((HANDLE)pthread_id);
+
+   if(EINA_UNLIKELY(sched_priority == THREAD_PRIORITY_TIME_CRITICAL))
+      {
+         sched_priority -= RTNICENESS;
+
+         /* We don't change the policy */
+         if (sched_priority < 1)
+            {
+               EINA_LOG_INFO("RT prio < 1, setting to 1 instead");
+               sched_priority = 1;
+            }
+         if(!SetThreadPriority((HANDLE)pthread_id, sched_priority))
+            {
+                EINA_LOG_ERR("Unable to query sched parameters");
+            } 
+     }
+     else
+      {
+         sched_priority += NICENESS;
+
+         /* We don't change the policy */
+         if (sched_priority > THREAD_PRIORITY_TIME_CRITICAL)
+            {
+               EINA_LOG_INFO("Max niceness reached; keeping max (THREAD_PRIORITY_TIME_CRITICAL)");
+               sched_priority = THREAD_PRIORITY_TIME_CRITICAL;
+            }
+         if(!SetThreadPriority((HANDLE)pthread_id, sched_priority))
+            {
+                EINA_LOG_ERR("Unable to query sched parameters");
+            } 
+      }
+
+
+}
+#endif
```

Comment:
Maybe, in this case, it makes sense to separate the two files creating a POSIX and a WIN32 implementation, what do you think?

Reply:
@JPTIZ what do you think?

Reply:
For me it makes sense. That would lead to a `_eina_sched_prio_drop` inline function in both .x files, which makes total sense :)

Reply:
Ok, i'll do it but I have 2 .c files (*_win32.c and *_posix.c) , not .x. Should I change it?